docs: Added and updated readmes across the project (#37)
EtienneDx/git-mentor: README.md

Reviewer: I guess this would be the file where we expect to see the badges of the repo (nb of contributors, build status, ci status, last update, etc), a logo and maybe the intent of the project. What do you think?

## Components

- [database](./database/): The shared database for the different git mentor applications
- [docs](./docs/): Documentation of the project
- [git-server](./git-server/): A simple git server library
- [git-server-example](./git-server-example/): A simple example of the use of the git-server library
- [gmt-api](./gmt-api/): The APIs of the project
- [gmt-cdk](./gmt-cdk/): The CDK stacks of the project
- [gmt-common](./gmt-common/): The common components of the project, reused by the different applications
- [gmt-server](./gmt-server/): The git server application
- [gmt-web-app](./gmt-web-app/): The front-end application of the project
- [ssh-server](./ssh-server/): A simple ssh server library, reused by the git-server library

## Setting up the development environment

### Prerequisites

#### Rust

The project uses Rust. To install Rust, follow the instructions at [https://www.rust-lang.org/tools/install](https://www.rust-lang.org/tools/install).

#### Node.js

> The front-end uses Node.js. To install Node.js, follow the instructions at [https://nodejs.org/en/download/](https://nodejs.org/en/download/). As of the 25/04/2024, the project uses v20.10

Reviewer: ```suggestion
The frontend uses Node.js. To install Node.js, follow the instructions at [https://nodejs.org/en/download/](https://nodejs.org/en/download/). As of the 25/04/2024, the project uses v20.10
```
front-end is a compound adjective 😇 [ref](https://medium.com/@ist.stevkovski/is-it-front-end-or-front-end-or-frontend-3ae717cae4aa). I suggest frontend or front end

#### AWS CDK

To generate the resources used by the project in AWS, you need to install the cdk:

```sh
npm install -g aws-cdk
```

#### Database

To run the database, it is recommended to use docker and the provided docker-compose scripts. To install docker, follow the instructions at [https://docs.docker.com/get-docker/](https://docs.docker.com/get-docker/).

> Otherwise, you would need to modify the environments of the project consuming the database. The database is a Postgre-SQL db.

Reviewer: ```suggestion
Otherwise, you would need to modify the environments of the project consuming the database. The database is a [Postgre-SQL](https://www.postgresql.org/) system.
```

### Running the project

1. Clone the repository:

```sh
git clone https://github.com/EtienneDx/git-mentor
```

2. Follow the instructions in each package.

Most backend (rust) projects can be runned directly using cargo:

```sh
cargo run
```

> The front-end uses npm:

Reviewer: same suggestion as above (there are a few other instances of it, but I leave it to you to find them)
```suggestion
The frontend uses npm:
```

```sh
npm install
npm run start
```

> */!\* the front-end requires a more complex setup to run properly, see the [gmt-web-app](./gmt-web-app/) documentation for more information on how to generate the APIs SDK.

Reviewer: This might lead to confusion. I would phrase it like such: "for the front end, go see the documentation, else use cargo run"
